include/api/wasmedge/wasmedge.h

```diff
@@ -2660,6 +2725,96 @@ WASMEDGE_CAPI_EXPORT extern WasmEdge_Result WasmEdge_VMRunWasmFromASTModule(
+/// by name.
+///
+/// This is the function to invoke a WASM function rapidly.
+/// Load and instantiate the WASM module from a buffer, and then invoke a
+/// function by name and parameters. If the `Returns` buffer length is smaller
+/// than the arity of the function, the overflowed return values will be
+/// discarded.
+///
+/// The caller owns the object and should call `WasmEdge_AsyncDelete` to free
+/// it.
+///
+/// \param Cxt the WasmEdge_VMContext.
+/// \param Buf the buffer of WASM binary.
+/// \param BufLen the length of the buffer.
+/// \param FuncName the function name WasmEdge_String.
+/// \param Params the WasmEdge_Value buffer with the parameter values.
+/// \param ParamLen the parameter buffer length.
+///
+/// \returns WasmEdge_Async. Call `WasmEdge_AsyncGet` for the result, and call
+/// `WasmEdge_AsyncDelete` to delete this object.
+WASMEDGE_CAPI_EXPORT extern WasmEdge_Async *WasmEdge_VMAsyncRunWasmFromBuffer(
+    WasmEdge_VMContext *Cxt, const uint8_t *Buf, const uint32_t BufLen,
+    const WasmEdge_String FuncName, const WasmEdge_Value *Params,
+    const uint32_t ParamLen);
+
+/// Instantiate the WASM module from a WasmEdge AST Module and asynchronous
+/// invoke a function by name.
+///
+/// This is the function to invoke a WASM function rapidly.
+/// Load and instantiate the WASM module from the WasmEdge AST Module, and then
+/// invoke the function by name and parameters. If the `Returns` buffer length
+/// is smaller than the arity of the function, the overflowed return values will
+/// be discarded.
+///
+/// The caller owns the object and should call `WasmEdge_AsyncDelete` to free
+/// it.
+///
+/// \param Cxt the WasmEdge_VMContext.
+/// \param ASTCxt the WasmEdge AST Module context generated by loader or
+/// compiler.
+/// \param FuncName the function name WasmEdge_String.
+/// \param Params the WasmEdge_Value buffer with the parameter values.
+/// \param ParamLen the parameter buffer length.
+///
+/// \returns WasmEdge_Async. Call `WasmEdge_AsyncGet` for the result, and call
+/// `WasmEdge_AsyncDelete` to delete this object.
+WASMEDGE_CAPI_EXPORT extern WasmEdge_Async *
+WasmEdge_VMAsyncRunWasmFromASTModule(WasmEdge_VMContext *Cxt,
+                                     const WasmEdge_ASTModuleContext *ASTCxt,
+                                     const WasmEdge_String FuncName,
+                                     const WasmEdge_Value *Params,
+                                     const uint32_t ParamLen);
+
+/// Load the WASM module from a WASM file.
+///
+/// This is the first step to invoke a WASM function step by step.
+/// Load and parse the WASM module from the file path. You can then call
+/// `WasmEdge_VMValidate` for the next step.
+///
+/// \param Cxt the WasmEdge_VMContext.
+/// \param Path the NULL-terminated C string of the WASM file path.
+///
+/// \returns WasmEdge_Result. Call `WasmEdge_ResultGetMessage` for the error
+/// message.
+
```

Comment:
I am not sure where does this section belong to?

Reply:
Removed.